frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-cpu-patches.ts

```diff
@@ -0,0 +1,37 @@
+import { PatchBuilder, PatchOperation } from '@console/shared/src/k8s';
+import { VMLikeEntityKind } from '../../../types';
+import { getVMLikePatches } from '../vm-template';
+import {
+  getCPU,
+  getResourcesRequestsCPUCount,
+  getResourcesLimitsCPUCount,
+  asVM,
+} from '../../../selectors/vm';
+
+export const getDedicatedCpuPatch = (
+  vmLikeEntity: VMLikeEntityKind,
+  dedicatedCpuPlacement: boolean,
+) => {
+  const vm = asVM(vmLikeEntity);
+  const isCPUAvailable = !!getCPU(vm);
+  const resourcesCPU = getResourcesRequestsCPUCount(vm) || getResourcesLimitsCPUCount(vm);
+
+  let newCPU;
+  if (resourcesCPU) {
+    newCPU = { dedicatedCpuPlacement };
+  } else {
+    newCPU = { cores: 1, dedicatedCpuPlacement };
+  }
+
+  const patches = [
+    new PatchBuilder(
```

Comment:
can we please branch this to new PatchBuilders to make the intent more clear? and to remove the ternary comparison duplication